add v180 release content (#62)
espressif/developer-portal: content/blog/vscode-esp-idf-v1.8.0/index.md

---
title: "Visual Studio Code ESP-IDF v1.8.0 extension"
date: 2024-07-12T16:07:09+08:00
showAuthor: false
tags: ["News", "ESP-IDF extension", "VSCode", "Visual Studio Code"]
authors:
  - "brian-ignacio"
---

## Introduction

> With the v1.8.0 of [Espressif ESP-IDF for Visual Studio Code extension](https://github.com/espressif/vscode-esp-idf-extension) we introduce a few new features and lot of bug fixes as described in [v1.8.0 Release](https://github.com/espressif/vscode-esp-idf-extension/releases/tag/v1.8.0). If you face any issue or improvement ideas please share them as a [Github issue](https://github.com/espressif/vscode-esp-idf-extension/issues).

Reviewer: It might be better to use VS Code in the extension name and elsewhere, because it is shorter and almost everybody know that it stands for Visual Studio Code anyway.

    default=true
    src="img/debug-example.webp"
    alt=""
    caption=""
    >}}

## Advanced configuration of Eclipse CDT Debug Adapter

There are many configurable options to customize the debug adapter if desired. You can view a deeper description in the [debugging documentation](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/DEBUGGING.md).

## ESP-IDF version switcher

Whenever you use the **ESP-IDF: Configure ESP-IDF Extension** to install ESP-IDF in your system, these installation are saved in our extension as global state IDF setup. You can now use the **ESP-IDF: Select Current ESP-IDF Version** to set in your current project (workspace folder in Visual Studio Code language) the ESP-IDF version you desire to use. If you want to delete these IDF setups from the extension global state you can use the **ESP-IDF: Clear Saved ESP-IDF Setups**.

{{< figure
    default=true
    src="img/idf-version-list.webp"
    alt=""
    caption=""
    >}}

The current ESP-IDF version for currently open ESP-IDF project is shown in the status bar items like this:

{{< figure
    default=true
    src="img/idf-version-status-bar.webp"
    alt=""
    caption=""
    >}}

> ## Add ESP-IDF: idf.py reconfigure task command

Reviewer: If you choose to update this section title according to my suggestion above, then:

```suggestion
## Use code navigation without full project build
```

For code navigation the [Microsoft C/C++ Extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools) or [Clangd extension](https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.vscode-clangd) can be used for C/C++ language support. These extensions provide feature such as syntax highlight, go to definition/declaration and other common language feature you would expect in an IDE. Many of these extensions rely on a `compile_commands.json` file, a JSON compilation database which consist of an array of “command objects”, where each command object specifies one way a translation unit is compiled in the project. More information about JSON compilation database can be found [here](https://clang.llvm.org/docs/JSONCompilationDatabase.html).

> The **ESP-IDF: idf.py reconfigure task** will execute ESP-IDF reconfigure task, which generate a compile_commands.json so the language extension can work properly. This is useful in case the user doesn't want to fully build the project but wants to see code navigation and other language features.

Reviewer: Please correct me if I am wrong, but, for now, the first paragraph reads like a long introduction to what code navigation is and how it works followed by a brief and simple introduction to what "ESP-IDF: idf.py reconfigure task" is. For those who know what is `compile_commands.json`, the second paragraph would be anticlamactic, as they would be wondering why they had to waste time on reading the first paragraph if the feature is so simple.

However, I am still not sure why the user might not want to fully build the project? Maybe you can provide good reasons for the readers to consider or to feel related?

I tried to rewrite this section with that idea in mind. Here is what I have got:

> Now you can generate a `compile_commands.json` file for code navigation extensions without fully building your project by using the command `ESP-IDF: idf.py reconfigure task`. Read on for more explanations.
>
> One of the key features of an IDE is code navigation. While working on your project, you usually want to benefit from such common language features as syntax highlight, code completion, go to definition/declaration, and others. For C/C++ language support, usually the extensions like [C/C++ extension][c-extension] or [clangd extension][clangd-extension] are used.
>
> Such extensions usually rely on a `compile_commands.json` file -- a JSON compilation database which consist of an array of *command objects*, where each command object specifies one way in which a translation unit is compiled in the project. More information about a JSON compilation database can be found [here][code-compilation-database].
>
> If, for any reason, you don’t want to fully build your project but still want to use code navigation, you can use the command `ESP-IDF: idf.py reconfigure task` to generate a `compile_commands.json` file for the language extension to work properly. This command executes the ESP-IDF reconfigure task, hence its name.

If you like it, here is the markdown version for it:

[c-extension]: https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools
[clangd-extension]: https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.vscode-clangd
[code-compilation-database]: https://clang.llvm.org/docs/JSONCompilationDatabase.html

```markdown
Now you can generate a `compile_commands.json` file for code navigation extensions without fully building your project by using the command `ESP-IDF: idf.py reconfigure task`. Read on for more explanations.

One of the key features of an IDE is code navigation. While working on your project, you usually want to benefit from such common language features as syntax highlight, code completion, go to definition/declaration, and others. For C/C++ language support, usually the extensions like [C/C++ extension][c-extension] or [clangd extension][clangd-extension] are used.

Such extensions usually rely on a `compile_commands.json` file -- a JSON compilation database which consist of an array of *command objects*, where each command object specifies one way in which a translation unit is compiled in the project. More information about a JSON compilation database can be found [here][code-compilation-database].

If, for any reason, you don’t want to fully build your project but still want to use code navigation, you can use the command `ESP-IDF: idf.py reconfigure task` to generate a `compile_commands.json` file for the language extension to work properly. This command executes the ESP-IDF reconfigure task, hence its name.

[c-extension]: https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools
[clangd-extension]: https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.vscode-clangd
[code-compilation-database]: https://clang.llvm.org/docs/JSONCompilationDatabase.html
```

> ## ESP-IDF Hints viewer support

Reviewer: I am sure the users will have these questions in mind:

- What is `ESP-IDF Hints` and how does it work? Where do the hints come from?
- How is it different from code navigation? Does it also rely on `compile_commands.json`?

This feature enhances your development experience by providing helpful hints for errors detected in your code generated from ESP-IDF during the build task.

In the source code editor, errors that match a line will be shown when the user mouse is hovering that line.

{{< video src="video/hints-hover" >}}

The ESP-IDF bottom panel automatically updates to display hints based on the errors in your currently opened file.

{{< figure
    src="img/hints-bottom-panel.webp"
    alt=""
    caption=""

>     >}}

Reviewer: Even if I click the zoom in button in the generated HTML, the text is still too small in the screenshot. Suggest updating the screen shot and making the text larger.

> You can manually search for hints by copy pasting errors with the **ESP-IDF: Search Error Hint**.

Reviewer: copy-pasting errors from where and to where? Even if you show it in the screenshot or video, it is still good to repeat it in writing to make it more discoverable.

---
title: "ESP-IDF Visual Studio Code Extension v1.8.0 release"
date: 2024-07-12T16:07:09+08:00
showAuthor: false
tags: ["News", "ESP-IDF extension", "VSCode", "Visual Studio Code"]
authors:
  - "brian-ignacio"
---

## Introduction

[ESP-IDF Visual Studio Code Extension](https://github.com/espressif/vscode-esp-idf-extension) v1.8.0 has been released! It introduces new features and many improvements. If you face any issue or improvement ideas please share them as a [Github issue](https://github.com/espressif/vscode-esp-idf-extension/issues).

Key features;

- **Eclipse CDT GDB Debug Adapter**

- **ESP-IDF version switcher**

- **Code Navigation without full project build**

- **ESP-IDF Hints viewer**

For other updates please review [v1.8.0 Release notes](https://github.com/espressif/vscode-esp-idf-extension/releases/tag/v1.8.0).

> 

Reviewer: Still suggest adding this to keep the momentum:

Now let's look into these features one by one.

{{< figure
    default=true
    src="img/debug-example.webp"
    alt=""
    caption=""
    >}}

## Advanced configuration of Eclipse CDT Debug Adapter

There are many configurable options to customize the debug adapter if desired. You can view a deeper description in the [debugging documentation](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/DEBUGGING.md).

## ESP-IDF version switcher

Whenever you install ESP-IDF in your system using the command `ESP-IDF: Configure ESP-IDF Extension`, the installations are saved in our extension as global state IDF setup. Now you can use `ESP-IDF: Select Current ESP-IDF Version` to set the ESP-IDF version you want to use in your current project (_workspace_ folder in VS Code terminology). If you want to delete these IDF setups from the extension global state, you can use `ESP-IDF: Clear Saved ESP-IDF Setups`.

{{< figure
    default=true
    src="img/idf-version-list.webp"
    alt=""
    caption=""
    >}}

The current ESP-IDF version for currently open ESP-IDF project is shown in the status bar items like this:

{{< figure
    default=true
    src="img/idf-version-status-bar.webp"
    alt=""
    caption=""
    >}}

> ## Use code navigation without full project build

Reviewer: Can you please add the information on why the user might not want to fully build the project? In which case it is good to do or what are the benefits?

You can either add it to the article yourself or drop some ideas in a comment and I will what to add to the article myself.

This is a question from a [previous comment](https://github.com/espressif/developer-portal/pull/62#discussion_r1685976723).

Author: Basically because doing a full build will require longer time and this reconfigure task is the first step to build but it is faster. On any regular build this command is actually run first to configure project and then ninja is used as build tool run the rest of the build as described in here: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/build-system.html#using-cmake-directly

Reviewer: Thanks! I will make respective updates in the following comments.

---
title: "ESP-IDF VS Code Extension v1.8.0 release"
date: 2024-07-12T16:07:09+08:00
showAuthor: false
tags: ["News", "ESP-IDF extension", "VSCode", "Visual Studio Code"]
authors:
  - "brian-ignacio"
---

## Introduction

[ESP-IDF VS Code Extension](https://github.com/espressif/vscode-esp-idf-extension) v1.8.0 has been released! It introduces new features and many improvements. If you face any issue or improvement ideas please share them as a [Github issue](https://github.com/espressif/vscode-esp-idf-extension/issues).

This article covers the following key features:

<!-- no toc -->
- [Eclipse CDT GDB Debug Adapter](#eclipse-cdt-gdb-debug-adapter)
- [ESP-IDF version switcher](#esp-idf-version-switcher)

> - [Code Navigation without full project build](#code-navigation-without-full-project-build)

Reviewer: Sorry, I missed this capitalization issue in my original suggestion.

```suggestion
- [Code navigation without full project build](#code-navigation-without-full-project-build)
```

- [ESP-IDF Hints viewer](#esp-idf-hints-viewer)

For other updates please review [v1.8.0 Release notes](https://github.com/espressif/vscode-esp-idf-extension/releases/tag/v1.8.0). Now let's look into these features one by one.

## Eclipse CDT GDB Debug Adapter

First, we will briefly talk about the implementation of debug adapters in VS Code and then will move on to the actual Eclipse CDT Debug Adapter.

### What is a debug adapter in VS Code?

> VS Code implements a generic (language-agnostic) _Debugger UI_ that is based on an abstract protocol. That protocol was specifically introduced to communicate with debugger backends. As debuggers typically use their own long-established protocols, some "adapter" is needed to allow debugger UI to communicate with debugger backends.

Reviewer: > Is it an actual term from VS Code docs? Suggest giving a link. If it is called differently in the VS Code context, suggest making the naming consistent.

This is from my [previous comment](https://github.com/espressif/developer-portal/pull/62#discussion_r1685917878).

Author: I think these sentences are taken as they are from vscode documentation regarding debug adapter.

Reviewer: Then maybe you can link to that document?

Author: done

This adapter is typically a standalone process with its own protocol. Both the Debug Adapter (DA) and the abstract protocol that is used between the VS Code's Debugger UI and DA -- the [Debug Adapter Protocol](https://microsoft.github.io/debug-adapter-protocol/) (DAP) -- have already been implemented by independent contributors. The diagram below illustrates it better:

{{< figure
    default=true
    src="img/debug-arch.webp"
    alt=""

>     caption=""

Reviewer: After you added the link to debugger UI docs, I saw that you borrowed the picture from the VS Code documentation. Suggest that we add a link to the source of the image to give credit to the original creator.:

```suggestion
    caption="The image is borrowed from [VS Code documentation](https://code.visualstudio.com/api/extension-guides/debugger-extension#debugging-architecture-of-vs-code)."
```

    alt=""
    caption=""
    >}}

## Advanced configuration of Eclipse CDT Debug Adapter

There are many configurable options to customize the debug adapter if desired. You can view a deeper description in the [debugging documentation](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/DEBUGGING.md).

## ESP-IDF version switcher

Whenever you install ESP-IDF in your system using the command `ESP-IDF: Configure ESP-IDF Extension`, the installations are saved in our extension as global state IDF setup. Now you can use `ESP-IDF: Select Current ESP-IDF Version` to set the ESP-IDF version you want to use in your current project (_workspace_ folder in VS Code terminology). If you want to delete these IDF setups from the extension global state, you can use `ESP-IDF: Clear Saved ESP-IDF Setups`.

{{< figure
    default=true
    src="img/idf-version-list.webp"
    alt=""
    caption=""
    >}}

The current ESP-IDF version for currently open ESP-IDF project is shown in the status bar items like this:

{{< figure
    default=true
    src="img/idf-version-status-bar.webp"
    alt=""
    caption=""
    >}}

## Code navigation without full project build

> Now you can generate a `compile_commands.json` file for code navigation extensions without fully building your project by using the command `ESP-IDF: idf.py reconfigure task`. Read on for more explanations.

Reviewer: ```suggestion
VS Code extensions for code navigation are dependent on a `compile_commands.json` file. Before this version, you could only generate such a file by fully building your project. Starting from this version, the file can be generated by running a dedicated command `ESP-IDF: idf.py reconfigure task`. It saves much time. Read on for more beginner-friendly explanation.
```
